runtime/parachains/src/hrmp/benchmarking.rs

```diff
@@ -0,0 +1,287 @@
+	let message_size = config.hrmp_channel_max_message_size;
+
+	let sender: ParaId = from.into();
+	let sender_origin: crate::Origin = from.into();
+
+	let recipient: ParaId = to.into();
+	let recipient_origin: crate::Origin = to.into();
+
+	let output = [(sender, sender_origin.clone()), (recipient, recipient_origin.clone())];
+
+	// Make both a parachain if they are already not.
+	if !Paras::<T>::is_parachain(sender) {
+		register_parachain_with_balance::<T>(sender, deposit);
+	}
+	if !Paras::<T>::is_parachain(recipient) {
+		register_parachain_with_balance::<T>(recipient, deposit);
+	}
+
+	assert_ok!(Hrmp::<T>::hrmp_init_open_channel(
+		sender_origin.clone().into(),
+		recipient,
+		capacity,
+		message_size
+	));
+
+	if matches!(until, ParachainSetupStep::Requested) {
+		return output
+	}
+
+	assert_ok!(Hrmp::<T>::hrmp_accept_open_channel(recipient_origin.into(), sender));
+	if matches!(until, ParachainSetupStep::Accepted) {
+		return output
+	}
+
+	Hrmp::<T>::process_hrmp_open_channel_requests(&Configuration::<T>::config());
+	if matches!(until, ParachainSetupStep::Established) {
+		return output
+	}
+
+	let channel_id = HrmpChannelId { sender, recipient };
+	assert_ok!(Hrmp::<T>::hrmp_close_channel(sender_origin.clone().into(), channel_id));
+	if matches!(until, ParachainSetupStep::CloseRequested) {
+		// NOTE: this is just for expressiveness, otherwise the if-statement is 100% useless.
+		return output
+	}
+
+	output
+}
+
+const PREFIX_0: u32 = 10_000;
+const PREFIX_1: u32 = PREFIX_0 * 2;
+const MAX_UNIQUE_CHANNELS: u32 = 128;
+
+static_assertions::const_assert!(MAX_UNIQUE_CHANNELS < PREFIX_0);
+
+frame_benchmarking::benchmarks! {
+	where_clause { where <T as frame_system::Config>::Origin: From<crate::Origin> }
+
+	hrmp_init_open_channel {
+		let sender_id: ParaId = 1u32.into();
+		let sender_origin: crate::Origin = 1u32.into();
+
+		let recipient_id: ParaId = 2u32.into();
+
+		// make sure para is registered, and has enough balance.
+		let deposit: BalanceOf<T> = Configuration::<T>::config().hrmp_sender_deposit.unique_saturated_into();
+		register_parachain_with_balance::<T>(sender_id, deposit);
+		register_parachain_with_balance::<T>(recipient_id, deposit);
+
+		let capacity = Configuration::<T>::config().hrmp_channel_max_capacity;
+		let message_size = Configuration::<T>::config().hrmp_channel_max_message_size;
+	}: _(sender_origin, recipient_id, capacity, message_size)
+	verify {
+		assert_last_event::<T>(
+			Event::<T>::OpenChannelRequested(sender_id, recipient_id, capacity, message_size).into()
+		);
+	}
+
+	hrmp_accept_open_channel {
+		let [(sender, _), (recipient, recipient_origin)] = establish_para_connection::<T>(1, 2, ParachainSetupStep::Requested);
+	}: _(recipient_origin, sender)
+	verify {
+		assert_last_event::<T>(Event::<T>::OpenChannelAccepted(sender, recipient).into());
+	}
+
+	hrmp_close_channel {
+		let [(sender, sender_origin), (recipient, _)] = establish_para_connection::<T>(1, 2, ParachainSetupStep::Established);
```

Comment:
Exceeds hard limit for the line width (as well as several other lines in this file)

Reply:
err yeah, cargo fmt is not working at all in macros unfortunately..